pg_search/src/postgres/scan.rs

```diff
@@ -55,15 +57,77 @@ pub extern "C" fn amrescan(
     let nkeys = nkeys as usize;
     let keys = unsafe { std::slice::from_raw_parts(keys as *const pg_sys::ScanKeyData, nkeys) };
 
-    // Convert the first scan key argument into a byte array. This is assumed to be the `::jsonb` search config.
-    let config_jsonb = unsafe {
-        JsonB::from_datum(keys[0].sk_argument, false)
-            .expect("failed to convert query to tuple of strings")
+    let search_config = match ScanStrategy::try_from(keys[0].sk_strategy).expect("invalid strategy")
```

Comment:
I think I'm not clear on how the user should pass `offset_rows` and `limit_rows` with this interface (assuming the user interacts with it directly, and not through `search`).

I can see how there's opportunity to pass those as fields in a `SearchConfigJson`, but I'm not sure about the `TextQuery` and `SearchQueryInput` variants.

Reply:
They don't!  By design.

If users want that then they need to stick with `.search()` or `.score_bm25()`.